### PR TITLE
Initial code

### PR DESCRIPTION
**Pull request by derins**
### Why:
* Added the initial code for the plot-viewer-lite.

### What:
* XY sorting is done, and the output grid is successfully produced.

### Checklist
- [x] Finished My Changes
- [ ] Automated Unit Tests

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    <label>Filter</label>
+      <select name="filter_type" id="filter_type">
+        <option disabled selected value> -Filter- </option>
+      </select>
+
+    <label>Normalization</label>
+      <select name="normalization" id="normalization">
+        <option disabled selected value> -Normalization- </option>
+        <option value="norm">Summer Total Peak Normalization</option>
+		    <option value="hour_norm">Hourly Normalization</option>
```
Comment (jongoncalves) [inline index.html]: Fix the indentation here.

File: index.html [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      </select>
+
+    <button type="button" onclick="sortPlots()">submit</button>
+    </form>
+
+    <div class="selected_options">
```
Comment (jongoncalves) [inline index.html]: Technically you don't need to change anything, but using hyphens for class names better IMHO. It prevents the ability to use is as a direct identifier in JS, which eliminates confusion I think. Also, there's a bunch of approaches on how to do this, however, one that I've found pretty successful is: http://getbem.com/introduction/
Just something to think about...

File: scripts/filter_type.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    $("#filter_type").empty().append('<option disabled selected value> -Filter- </option>')
+
+    var y_axis = $("#y_axis_search");
+    var x_axis = $("#x_axis_search");
+    var locContent = "<option disabled selection value> - Location - </option><option> Atlanta_GA </option><option> Baltimore_MD </option><option> Boston_MA </option><option> Calgary_AB </option><option> Chicago_IL </option><option> Dallas_TX </option><option> Houston_TX </option><option> Miami_FL </option><option> Minneapolis_MN </option><option> Nashville_TN </option><option> NewYorkCity_NY </option><option> QuebecCity_QC </option><option> SanAntonio_TX </option><option> Toronto_ON </option>"
+    var seasContent = "<option disabled selection value> - Season - </option><option>Winter</option><option>Spring</option><option>Summer</option>";
+    var mtContent = "<option disabled selection value> - Mix Type - </option><option>URBAN</option><option>MIXED</option><option>SUBURBAN</option><option>RURAL</option>";
+
+
+    $("#filter_type").empty().append('<option disabled selected value> -Filter- </option>')
```
Comment (jongoncalves) [inline scripts/filter_type.js]: Why this is called twice? It's being called a couple of lines above already.

File: scripts/filter_type.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    if (x_axis.val() == "location") {
+      if (y_axis.val() == "season"){
+        $("#filter_type").html(mtContent);
```
Comment (jongoncalves) [inline scripts/filter_type.js]: Build all the content that is being passed into the html function once, as a string, and then change the DOM only once at the end. Changing the DOM is expensive.

File: scripts/filter_type.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        }
+      if (y_axis.val() == "mix_type"){
+        $("#filter_type").html(seasContent)
+      }
+    }
+    else if (x_axis.val() == "season") {
+      if (y_axis.val() == "location"){
+        $("#filter_type").html(mtContent);      }
```
Comment (jongoncalves) [inline scripts/filter_type.js]: Trailing bracket should be in a new line.

File: scripts/main.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
+function sortPlots() {
+
+    var x_axis = $('#x_axis_search').find(":selected").val();
+    var y_axis = $('#y_axis_search').find(":selected").val();
+    var filter = $('#filter_type').find(":selected").val();
+    console.log(filter)
```
Comment (jongoncalves) [inline scripts/main.js]: Remove log statement.

File: scripts/filter_type.js (hunk 4 above)
Comment (jongoncalves) [reply scripts/filter_type.js]: Also, cache the filter type jquery obj in a var.

Comment (jongoncalves) [reply scripts/filter_type.js]: You can also cache the axis vals

File: scripts/main.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+    var rows = [];
+    var columns = [];
+    var filtered = [];
+    var xy_sort = [];
+
+    $(".selected_options").empty().append("<h3> x-axis: " + x_axis_text + "</h3>");
+    $(".selected_options").append("<h3> y-axis: " + y_axis_text + "</h3>");
+    $(".selected_options").append("<h3> filter: " + filter_text + "</h3>");
+    $(".selected_options").append("<h3> nomalization: " + normalization_text + "</h3>");
```
Comment (jongoncalves) [inline scripts/main.js]: Same as above, build a string and append to the DOM only once.

File: scripts/filter_type.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    var y_axis = $("#y_axis_search");
+    var x_axis = $("#x_axis_search");
+    var locContent = "<option disabled selection value> - Location - </option><option> Atlanta_GA </option><option> Baltimore_MD </option><option> Boston_MA </option><option> Calgary_AB </option><option> Chicago_IL </option><option> Dallas_TX </option><option> Houston_TX </option><option> Miami_FL </option><option> Minneapolis_MN </option><option> Nashville_TN </option><option> NewYorkCity_NY </option><option> QuebecCity_QC </option><option> SanAntonio_TX </option><option> Toronto_ON </option>"
+    var seasContent = "<option disabled selection value> - Season - </option><option>Winter</option><option>Spring</option><option>Summer</option>";
+    var mtContent = "<option disabled selection value> - Mix Type - </option><option>URBAN</option><option>MIXED</option><option>SUBURBAN</option><option>RURAL</option>";
+
+
+    $("#filter_type").empty().append('<option disabled selected value> -Filter- </option>')
+
+    if (x_axis.val() == "location") {
```
Comment (jongoncalves) [inline scripts/filter_type.js]: Use `===` instead of `==` everywhere.

File: scripts/main.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    // filter type being determined
+    // filter options are displayed through filter_type.js
+    var sort_options = ["season", "mix_type", "location"];
+    sort_options.splice(sort_options.indexOf(x_axis), 1);
+    sort_options.splice(sort_options.indexOf(y_axis), 1);
+    var filter_type = sort_options[0];
+
+
+    // accesses the manually updated json file on s3 bucket
+    $.getJSON("https://s3-us-west-1.amazonaws.com/lctk.data/derin-test/2_plots_list.json", function(plots) {
```
Comment (jongoncalves) [inline scripts/main.js]: Put this URL in a variable for readability.

File: scripts/main.js [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+            $("#content").empty();
+            for (var x = 0; x < xy_sort.length; x++) {
+                $("#content").append("<div id='rows' class='row" + x + "'>");
+                for (var y = 0; y < xy_sort[x].length; y++) {
+                    $(".row" + x).append("<div class='plots'>");
+                    $(".row" + x).append("<img src=" + '"' + xy_sort[x][y].model_img_url + '"' + "style='width:100%>'" + "</br>");
+                    $(".row" + x).append("<a href=" + '"' + xy_sort[x][y].csv_url + '"' + "download target='_blank'><i class='fa fa-download'></i> Download CSV</a>");
+                    $(".row" + x).append("</div>");
+                }
+                $("#content").append("</div>");
```
Comment (jongoncalves) [inline scripts/main.js]: Same append comments as above.

File: styles.css [hunk 11]
```diff
@@ -0,0 +1,15 @@
+* {
+  box-sizing: border-box;
+}
+
+#rows {
+    overflow: visible;
+    white-space:nowrap;
```
Comment (jongoncalves) [inline styles.css]: Add a whitespace after `:`

File: scripts/main.js [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+            }
+        // error message
+        } else {
+          $("#content").empty().append("<h2>please select a different value for each dropdown and make sure each drop down has a selected value</h2>")
```
Comment (jongoncalves) [inline scripts/main.js]: Be consistent with your `;`, either use it always or never.

File: scripts/main.js [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+            }
+            // sorts array acvcording to y-axis input
+            filtered.sort((a, b) => (a[y_axis] > b[y_axis]) ? 1 : -1);
+            //creates new arrays - rows - of objects with unique y-axis input values
+            rows.push(filtered[0]);
+            // adds all the rows in to the final array - xy_sort - which is an array of arrays of objects
+            for (var plot = 0; plot < filtered.length - 1; plot++) {
+                if (filtered[plot][y_axis] == filtered[plot + 1][y_axis]) {
+                    rows.push(filtered[plot + 1]);
+                } else {
```
Comment (jongoncalves) [inline scripts/main.js]: Be consistent with your `else`s and `else if`s, sometimes they are inline with the previous bracket and sometimes not. Make them all inline.

File: index.html (hunk 1 above)
Comment (derins) [reply index.html]: Fixed

File: index.html (hunk 2 above)
Comment (derins) [reply index.html]: Converted the class and id names to hyphens. Will look in to Block Element Modifier.

File: scripts/filter_type.js (hunk 3 above)
Comment (derins) [reply scripts/filter_type.js]: Fixed

File: scripts/filter_type.js (hunk 4 above)
Comment (derins) [reply scripts/filter_type.js]: I haven't cached the values yet, will look into it.

File: scripts/filter_type.js (hunk 5 above)
Comment (derins) [reply scripts/filter_type.js]: Fixed

File: scripts/main.js (hunk 9 above)
Comment (derins) [reply scripts/main.js]: Fixed

File: scripts/main.js (hunk 10 above)
Comment (derins) [reply scripts/main.js]: Fixed

File: styles.css (hunk 11 above)
Comment (derins) [reply styles.css]: Fixed

File: scripts/main.js (hunk 12 above)
Comment (derins) [reply scripts/main.js]: I fixed it to be using ; on every line. Do you usually prefer using it or not?

File: scripts/main.js (hunk 13 above)
Comment (derins) [reply scripts/main.js]: Fixed

File: scripts/main.js (hunk 12 above)
Comment (jongoncalves) [reply scripts/main.js]: I prefer to use `;` but don't really care as long as it's consistent.

File: index.html [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+  <nav class="navbar navbar-default">
+    <div class="container-fluid">
+      <div class="navbar-header">
+        <a class="navbar-brand" href="">Viewer</a>
+      </div>
+    </div>
+  </nav>
+
+  <form>
+    <div id = "sort-form">
```
Comment (jongoncalves) [inline index.html]: no spaces around `=`

File: index.html [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+    </form>
+
+    <div class="selected-options">
+    </div>
+
+    <div id="content">
+    </div>
+
+    <script src="https://code.jquery.com/jquery-1.10.1.js"></script>
+    <script src ="https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
```
Comment (jongoncalves) [inline index.html:43]: no space before `=`

File: scripts/config.js [hunk 16]
```diff
@@ -0,0 +1,3 @@
+var sort_options = ["season", "mix_type", "location"];
+var json_url = "https://s3-us-west-1.amazonaws.com/lctk.data/derin-test/2_plots_list.json";
+var filters = ["normalization"];
```
Comment (jongoncalves) [inline scripts/config.js]: Don't put these variables in a separate files, they should be in the file they get used in. It causes confusion and people may not be aware where it get loaded. Plus the skews the scope of them.

File: scripts/filter_type.js [hunk 17]
```diff
@@ -0,0 +1,82 @@
+//used for the filter type to be determined once x_axis and y_axis are selected
+
+$(document).ready(function() {
+    var plots = [];
+
+    $("#y-axis-search, #x-axis-search").change(function() {
+        $("#filter-type").empty().append('<option disabled selected value> -Filter- </option>')
```
Comment (jongoncalves) [inline scripts/filter_type.js]: missing `;`

File: scripts/filter_type.js [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+        $.getJSON(json_url, function(plots) {
+
+            var y_axis = $("#y-axis-search");
+            var x_axis = $("#x-axis-search");
+            var dist_options = [];
```
Comment (jongoncalves) [inline scripts/filter_type.js:16]: be consistent with your naming. For javascript, `camelCase` your names, not `kebab-case`

File: scripts/filter_type.js [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+                        lookup[name] = 1;
+                        result.push(name);
+                    }
+                }
+                result.sort();
+                dist_options.push(result);
+            }
+            //dist_options includes distinct values of sort options as seperate arrays of strings
+
+            $("#filter-type").empty().append('<option disabled selected value> -Filter- </option>');
```
Comment (jongoncalves) [inline scripts/filter_type.js]: This is being called up above, why is it still being called here? Seems like you should be able to consolidate/simplify this.

File: scripts/main.js [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+                    row_str += "<img src=" + '"' + xy_sort[x][y].model_img_url + '"' + "style='width:100%>'" + "</br>";
+                    row_str += "<a href=" + '"' + xy_sort[x][y].csv_url + '"' + "download target='_blank'><i class='fa fa-download'></i> Download CSV</a>";
+                    row_str += "</div>";
+                    $(".row" + x).append(row_str);
+                }
+                $("#content").append("</div>");
+            }
+        }
+        // error message
+        else {
```
Comment (jongoncalves) [inline scripts/main.js]: I thought you chose to inline these instead of new lining them

File: scripts/filter_type.js [hunk 21]
```diff
@@ -0,0 +1,82 @@
...
+                    }
+                }
+                if (y_axis.val() === sort_options[2]) {
+                    filter_str = ("<option disabled selection value> - " + sort_options[1] + "</option>");
+                    for (var b = 0; b < dist_options[1].length; b++){
+                        filter_str += ("<option>" + dist_options[1][b] + "</option>");
+                    }
+                }
+            }
+            else if (x_axis.val() === sort_options[1]) {
```
Comment (jongoncalves) [inline scripts/filter_type.js]: this else conditional should be inlined

File: scripts/main.js [hunk 22]
```diff
@@ -0,0 +1,121 @@
+var plots;
```
Comment (jongoncalves) [inline scripts/main.js:1]: why is this variable scoped outside of the function?

File: scripts/main.js [hunk 23]
```diff
@@ -0,0 +1,121 @@
...
+function sortOptions() {
+
+    var x_axis_str = "<option disabled selected value> - select x-axis - </option>";
```
Comment (jongoncalves) [inline scripts/main.js:6]: same comment as before. Variable naming....

File: scripts/main.js [hunk 24]
```diff
@@ -0,0 +1,121 @@
...
+    $("#y-axis-search").empty().append(y_axis_str);
+
+    $.getJSON(json_url, function(plot) {
+        plots = plot;
+        var filters_str = "";
+        var lookup = {};
+        var result = [];
+
+        for (var l = 0; l < filters.length; l++){
+            filters_str += "<label>Filter" + (l+2) + "</label>";
```
Comment (jongoncalves) [inline scripts/main.js]: proper spacing around `+`

File: scripts/main.js [hunk 25]
```diff
@@ -0,0 +1,121 @@
...
+            filters_str += "<select id='filter" + (l+2) + "'>";
+            filters_str += "<option disabled selected value> - " + filters[l] + " - </option>";
+            lookup = {};
+            result = [];
+            for (var item, i = 0; item = plots[i++];) {
+                console.log("hey");
```
Comment (jongoncalves) [inline scripts/main.js]: remove `console`

File: scripts/main.js [hunk 26]
```diff
@@ -0,0 +1,121 @@
...
+    for (var v = 0; v < filters.length; v++){
+        selected_options += "<h3> " + filters[v] + ": " + filters_arr[v] + "</h3>";
+    }
+    $(".selected-options").empty().append(selected_options);
+    // filter type being determined
+    // filter options are displayed through filter_type.js
+    sort_options.splice(sort_options.indexOf(x_axis), 1);
+    sort_options.splice(sort_options.indexOf(y_axis), 1);
+    var filter_type = sort_options[0];
+        if ((x_axis != y_axis)) {
```
Comment (jongoncalves) [inline scripts/main.js]: Indentation is wrong here and you should use `!==` instead of `!=`

File: scripts/main.js [hunk 27]
```diff
@@ -0,0 +1,121 @@
...
+          // collects all the objects with the corresponding filter type in array - filtered
+            for (var filt = 0; filt < filters_arr.length; filt++) {
+                for (var plot = 0; plot < plots.length; plot++) {
+                    if ((plots[plot][filter_type] == filter) && (plots[plot][filters[0]] == filters_arr[filt])) {
```
Comment (jongoncalves) [inline scripts/main.js]: `===` instead of `==`

**Review by jongoncalves — COMMENTED** [review]
Don't make these sweeping changes to a PR. Fix the things outlined and let's get it in a state that makes sense, merge it in to master, then create a new branch and go through the whole process again otherwise this can go on forever.

File: index.html (hunk 14 above)
Comment (derins) [reply index.html]: fixed

File: index.html (hunk 15 above)
Comment (derins) [reply index.html:43]: fixed

File: scripts/config.js (hunk 16 above)
Comment (derins) [reply scripts/config.js]: fixed

File: scripts/filter_type.js (hunk 17 above)
Comment (derins) [reply scripts/filter_type.js]: fixed

File: scripts/filter_type.js (hunk 18 above)
Comment (derins) [reply scripts/filter_type.js:16]: fixed

File: scripts/main.js (hunk 23 above)
Comment (derins) [reply scripts/main.js:6]: fixed

File: scripts/main.js (hunk 24 above)
Comment (derins) [reply scripts/main.js]: fixed

File: scripts/main.js (hunk 25 above)
Comment (derins) [reply scripts/main.js]: fixed

File: scripts/main.js (hunk 26 above)
Comment (derins) [reply scripts/main.js]: fixed

File: scripts/main.js (hunk 27 above)
Comment (derins) [reply scripts/main.js]: fixed
